### DISCUSSION TITLE
NGINX Proxy Manager / NGINX Snippets not consistent & clear

### DISCUSSION THREAD

**Discussion by sarpuser** [authelia/authelia#7322] [Documentation Feedback]
The documentation for the [NGINX Proxy Manager](https://www.authelia.com/integration/proxies/nginx/#assumptions) assumes that the snippets are in /snippets/. The documentation for [NGINX](https://www.authelia.com/integration/proxies/nginx/#assumptions) assumes the snippets are in /config/nginx/snippets. When the NPM documentation says look at the NGINX docs to copy the configuration files, you end up copying files with paths that do not with with the [NPM advanced settings](https://www.authelia.com/integration/proxies/nginx-proxy-manager/#authelia-portal).

Two other side points:
1. It is not clear in the documentation of either NPM nor NGINX that the /snippets directory is in reference to those containers, not in reference to Authelia. If someone does not understand Docker very well, this may confuse them a bit.
2. The quick setup example on NPM's website uses /data instead of /config. I know the Authelia docs are examples, but I think it would be useful to change from /snippets or /config/snippets to /data/snippets.

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

It appears there is a discrepancy in the documentation regarding the location of configuration snippets between NGINX and NGINX Proxy Manager (NPM) when integrating with Authelia. This can indeed cause confusion, especially for users who are not deeply familiar with Docker or the specific directory structures used by these containers.

1. **Directory Path Differences**: The NGINX documentation assumes that snippets are located in `/config/nginx/snippets`, while the NPM documentation refers to them simply as being in `/snippets/`. When following the NPM documentation which suggests referring to the NGINX documentation, you should adjust the file paths accordingly to match the expected directory structure of NPM. This means changing any instance of `/config/nginx/snippets/` to `/snippets/` in the configuration files when setting them up in NPM.

2. **Clarification on Directory References**: It's important to note that the directory paths mentioned (`/snippets`, `/config/nginx/snippets`) are relative to the container's filesystem, not the host filesystem. This is a common source of confusion for users unfamiliar with how Docker handles volumes and file paths. Documentation could be improved by explicitly stating that these paths are internal to the containers.

3. **Suggestion on Standardizing Paths**: Your suggestion to standardize the use of `/data/snippets` across documentation to align with quick setup examples on NPM's website is thoughtful. It would help in maintaining consistency and reducing confusion. This change would involve updating all relevant documentation to reflect this new standard path.

For now, when setting up NPM with Authelia, ensure you adjust snippet paths as per the NPM's expected directory structure. If you encounter issues, double-check the container's actual directory structure either by accessing the container's shell or reviewing its Dockerfile or docker-compose definitions to confirm the correct paths.

If these discrepancies are causing setup issues, it might be beneficial to reach out to the maintainers of the Authelia and NPM documentation to suggest these clarifications and improvements.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Reply by sarpuser** [reply]
> it might be beneficial to reach out to the maintainers of the Authelia

Yeah thanks I hadn't thought of that!
